trustd/src/main.rs

```diff
@@ -25,119 +23,52 @@ pub enum Command {}
-            use postgresql_embedded::Settings;
-
-            let current_dir = env::current_dir()?;
-            let work_dir = current_dir.join(".trustify");
-            let db_dir = work_dir.join("postgres");
-            let settings = Settings {
-                username: self.database.username.clone(),
-                password: self.database.password.clone(),
-                temporary: false,
-                installation_dir: db_dir.clone(),
-                ..Default::default()
-            };
-
-            let mut postgresql = PostgreSQL::new(PostgreSQL::default_version(), settings);
-            postgresql.setup().await?;
-            postgresql.start().await?;
-
-            let port = postgresql.settings().port;
-            self.database.port = port;
-            self.creation = CreationMode::Bootstrap;
-
-            managed_db.replace(postgresql);
-
-            println!("postgresql installed under {:?}", db_dir);
-            println!("running on port {}", port);
-        }
-
-        let mut handles = JoinSet::new();
-
-        if self.with_http {
-            let http = trustify_server::Run {
-                database: self.database.clone(),
-                storage: self.storage.clone(),
-                creation: self.creation,
-                devmode: self.devmode,
-                infra: self.infra.clone(),
-                auth: self.auth.clone(),
-                http: self.http.clone(),
-                swagger_ui_oidc: self.swagger_ui_oidc.clone(),
-            };
-
-            handles.spawn_local(http.run());
-        }
-
-        while let Some(result) = handles.join_next().await {
-            match result {
-                Ok(result) => match result {
-                    Ok(_) => {}
-                    Err(err) => {
-                        log::error!("error {:?}", err);
-                    }
-                },
-                Err(err) => {
-                    log::error!("fundamental error {:?}", err);
+    async fn run_command(self) -> anyhow::Result<ExitCode> {
+        match self.command {
+            Some(Command::Api(run)) => run.run().await,
+            Some(Command::Db(run)) => run.run().await,
+            None => {
+                let Some(Command::Db(mut db)) =
+                    Trustd::parse_from(["trustd", "db", "migrate"]).command
+                else {
+                    unreachable!()
+                };
+                let postgres = db.start().await?;
+                if postgres.database_exists(&db.database.name).await.is_err() {
+                    db.command = db::Command::Create;
                 }
+                db.run().await?;
+
+                let api = Trustd::parse_from([
+                    "trustd",
+                    "api",
+                    "--auth-disabled",
+                    "--db-port",
+                    &postgres.settings().port.to_string(),
+                ]);
+                Ok(tokio::task::spawn_local(api.run()).await?)
```

Comment:
We'd probably want to embody this different, if we're also spawning the importer loop and wtf knows what else in the future?

Reply:
The importer loop is already "running". It's data driven, so if a PM adds an importer via the instructions in the README, it'll run each time they fire up trustd, until they delete it, because now the data won't be destroyed with each run.